Web test

andresmadeddie opened the pull request:
Install selenium server and Chromedriver
Configure config.properties
Create three Page Object classes Login, Product, and Cart
Create one component class
Create one util class for authentication
Create one abstract class
Add default username and password in testdata.properties
Create SauceDemoTest Class
Test successful
Add Web Test Cases file on the README.md

andresmadeddie committed:
rename methods

andresmadeddie committed:
xml suite

andresmadeddie committed:
remove target folder

andresmadeddie committed:
Install selenium server and Chromedriver
Configure config.properties
Create three Page Object classes Login, Product, and Cart
Create one component class
Create one util class for authentication
Create one abstract class
Add default username and password in testdata.properties
Create SauceDemoTest Class
Test successful
Add Web Test Cases file on the README.md

andresmadeddie committed:
Install selenium server and Chromedriver
Configure config.properties
Create three Page Object classes Login, Product, and Cart
Create one component class
Create one util class for authentication
Create one abstract class
Add default username and password in testdata.properties
Create SauceDemoTest Class
Test successful
Add Web Test Cases file on the README.md

andresmadeddie committed:
Install selenium server and Chromedriver
Configure config.properties
Create three Page Object classes Login, Product, and Cart
Create one component class
Create one util class for authentication
Create one abstract class
Add default username and password in testdata.properties
Create SauceDemoTest Class
Test successful
Add Web Test Cases file on the README.md

ankochergin commented on src/main/java/com/solvd/carina/demo/gui/pages/saucedemo/CartPage.java:
```diff
@@ -0,0 +1,34 @@
+        super(driver);
+        setPageOpeningStrategy(PageOpeningStrategy.BY_ELEMENT);
+        setUiLoadedMarker(pageTitleLabel);
+//        setPageAbsoluteURL(R.CONFIG.get("url"));
```
please remove if not needed

ankochergin commented on src/main/java/com/solvd/carina/demo/gui/pages/saucedemo/ProductsPage.java:
```diff
@@ -0,0 +1,43 @@
+        super(driver);
+        setPageOpeningStrategy(PageOpeningStrategy.BY_ELEMENT);
+        setUiLoadedMarker(pageTitleLabel);
+//        setPageAbsoluteURL(R.CONFIG.get("url"));
```
please remove if not needed

ankochergin commented on src/test/java/com/solvd/carina/demo/SauceDemoTest.java:
```diff
@@ -0,0 +1,50 @@
+
+    @Test
+    @MethodOwner(owner = "Andres")
+    public void LoginTest() {
```
```suggestion
    public void loginTest() {
```

andresmadeddie committed:
Merge pull request #1 from AndresSolvd/modifications

Modifications

AndresSolvd committed:
removed unnecessary code and comments, renamed loginTest

AndresSolvd committed:
Merge remote-tracking branch 'origin/WebTest' into WebTest

# Conflicts:
#	src/main/java/com/solvd/carina/demo/gui/pages/saucedemo/CartPage.java
#	src/main/java/com/solvd/carina/demo/gui/pages/saucedemo/LoginPage.java
#	src/main/java/com/solvd/carina/demo/gui/pages/saucedemo/ProductsPage.java
#	src/test/java/com/solvd/carina/demo/SauceDemoTest.java

AndresSolvd committed:
created AboutPage class
created BurgerMenu component
added burgerMenu list and method in productPage

andresmadeddie committed:
remove BurgerMenu.java
Add field and method for about in ProductsPage
Add test of about in test class
change xpath on AboutPage

AndresSolvd committed:
Test logout button

AndresSolvd committed:
change ip address to localhost

AndresSolvd committed:
create CheckOutYourInformationPage and test method

AndresSolvd committed:
create LeftSideMenu.java component and make changes accordingly

AndresSolvd committed:
Add TestWithThreads.java

jinocencio-solvd commented on src/test/java/com/solvd/carina/demo/SauceDemoTest.java:
```diff
@@ -0,0 +1,82 @@
+    @Test
+    @MethodOwner(owner = "Andres")
+    public void productAddedToTheCartTest() {
+        String productTitle = "Sauce Labs Onesie";
```
maybe use dataProvider?

jinocencio-solvd commented on src/main/java/com/solvd/carina/demo/gui/saucedemo/ProductsPage.java:
```diff
@@ -0,0 +1,57 @@
+        return products;
+    }
+
+    public CartPage clickCartButton() {
```
i think this method name could be more descriptive like openCartPage or getCartPage

jinocencio-solvd commented on src/test/java/com/solvd/carina/demo/TestWithThreads.java:
```diff
@@ -0,0 +1,79 @@
+
+import java.util.concurrent.*;
+
+public class TestWithThreads {
```
Have you tried configuring Maven/Testng or Carina configs to run methods  in parallel?

jinocencio-solvd commented on src/test/java/com/solvd/carina/demo/SauceDemoTest.java:
```diff
@@ -0,0 +1,82 @@
+        loginPage.typeUsername(username);
+        loginPage.typePassword(password);
+        ProductsPage productsPage = loginPage.clickLoginButton();
+        Assert.assertTrue(productsPage.isPageOpened(), "product page is not opened");
```
there is also assertPageOpened method accessible within AbstractPage class

jinocencio-solvd commented on README.md:
````diff
@@ -0,0 +1,23 @@
+```
+## Test Cases
+
+| Feature Owner (Author) | #   | Name                   | Preconditions                                                                                                  | Step # | Description                                            | Expected Result        |
````
Nice job with the table display!

AndresSolvd committed:
add product name to test case and modify test class accordingly
Change to assertPageOpened method

AndresSolvd committed:
added SauceDemoTest on web.xml

andresmadeddie committed:
added data-provider-thread-count="5" to web.xml

AndresSolvd committed:
Removed TestWithThreads and modified to  thread-=count the web.xml file

ankochergin commented on src/main/java/com/solvd/carina/demo/gui/saucedemo/components/LeftSideMenu.java:
```diff
@@ -0,0 +1,38 @@
+    public LoginPage clickLogOutButtonInLeftSideMenu() {
+        logOutButtonInLeftSideMenu.click();
+        return new LoginPage(driver);
+    }
```
it's already in LeftSideMenu so it's not needed to write so long names for methods and variables
```suggestion
    public LoginPage clickLogOutButton() {
        logOutButton.click();
        return new LoginPage(driver);
    }
```

ankochergin commented on src/main/java/com/solvd/carina/demo/gui/saucedemo/components/LeftSideMenu.java:
```diff
@@ -0,0 +1,38 @@
+    public AboutPage clickAboutButtonInLeftSideMenu() {
+        aboutButtonInLeftSideMenu.click();
+        return new AboutPage(driver);
+    }
```
same as in previous comment

ankochergin commented on src/test/java/com/solvd/carina/demo/SauceDemoTest.java:
```diff
@@ -0,0 +1,81 @@
+
+        LoginPage loginPage = new LoginPage(getDriver());
+        loginPage.open();
+        loginPage.assertPageOpened();
```
I prefer to write assertTrue(page.isPageOpened(), "XXX page is not opened");
It helps to customise message and code is more readable

ankochergin commented on src/test/resources/testng_suites/web.xml:
```diff
@@ -17,6 +17,7 @@
 	<test name="GSM arena web tests">
 		<classes>
 			<class name="com.solvd.carina.demo.WebSampleTest" />
+			<class name="com.solvd.carina.demo.SauceDemoTest"/>
```
specify each test method individually using <include> node to make it easier to support

AndresSolvd committed:
shorted names of LeftSideMenu methods and fields
change assert method to the previously Assert.assertTrue in the SauceDemoTest
modify web.xml to include each method test

andresmadeddie committed:
added checkOutTest
create new pages
added to web.xml
test successful
